Rakefile

```diff
@@ -0,0 +1,133 @@
+require "fileutils"
+require "rototiller"
+require "yard"
+require "rubocop/rake_task"
+require "flog_task"
+require "flay_task"
+require "roodi_task"
+require "rubycritic/rake_task"
+
+YARD_DIR = "doc".freeze
+DOCS_DIR = "docs".freeze
+
+task :default do
+  sh %(rake -T)
+end
+
+namespace :gem do
+  require "bundler/gem_tasks"
+end
+
+task yard: :"docs:yard"
+
+# rubocop:disable Metrics/BlockLength
+namespace :docs do
+  # docs:yard task
+  YARD::Rake::YardocTask.new
+
+  desc "Clean/remove the generated YARD Documentation cache"
+  task :clean do
+    original_dir = Dir.pwd
+    Dir.chdir(File.expand_path(File.dirname(__FILE__)))
+    sh "rm -rf #{YARD_DIR}"
+    Dir.chdir(original_dir)
```

Comment:
So I assume all this is to make sure you aren't in the directory that should be deleted when you run the clean up task.  But if you are, won't the chdir at the end fail?  Maybe check if it exists before chdiring into it.

Reply:
yep

Reply:
@RandellP i remember now why this stuff is here.  rake can be run from any dir, and it goes and finds the Rakefile, e.g.  in `../`
in which case we'd be in the wrong dir, and possibly create/delete files we didn't intend

Reply:
I hadn't thought about __FILE__ not working... good point there.
But my concern was what if the working dir was the YARD_DIR.  It would chdir out, delete it, then try to chdir back into the now deleted dir.

Reply:
ah.  good concern there too, but for a small feature.  i'll try to protect against it in a future commit